services/spar/test-integration/Test/Spar/AppSpec.hs

```diff
@@ -0,0 +1,159 @@
+              Servant.errReasonPhrase outcome `shouldBe` "success"
+              Servant.errBody outcome `shouldBe` mempty
+              uriScheme loc `shouldBe` (URI.Scheme "wire")
+              List.lookup "label"  qry `shouldBe` Nothing
+              List.lookup "userid" qry `shouldBe` (Just . cs . show $ uid)
+              List.lookup "cookie" qry `shouldNotBe` Nothing
+              List.lookup "cookie" qry `shouldNotBe` Just "$cookie"
+              let Just (ckies :: SBS) = List.lookup "cookie" qry
+                  cky :: SetCookie = parseSetCookie ckies
+              setCookieName cky `shouldBe` "zuid"
+              ('s', setCookieSecure cky) `shouldBe` ('s', False)  -- we're in integration test mode, no https here!
+              ('h', setCookieHttpOnly cky) `shouldBe` ('h', True)
+
+
+mkAuthnReqWeb :: SAML.IdPId -> TestSpar ResponseLBS
+mkAuthnReqWeb idpid = do
+  env <- ask
+  call $ get ((env ^. teSpar) . path (cs $ "/sso/initiate-login/" -/ SAML.idPIdToST idpid) . expect2xx)
+
+mkAuthnReqMobile :: SAML.IdPId -> TestSpar ResponseLBS
+mkAuthnReqMobile idpid = do
+  env <- ask
+  let succurl   = [uri|wire://login-granted/?cookie=$cookie&userid=$userid|]
+      errurl    = [uri|wire://login-denied/?label=$label|]
+      mk        = Builder.toLazyByteString . urlEncode [] . serializeURIRef'
+      arQueries = cs $ "success_redirect=" <> mk succurl <> "&error_redirect=" <> mk errurl
+      arPath    = cs $ "/sso/initiate-login/" -/ SAML.idPIdToST idpid <> "?" <> arQueries
+  call $ get ((env ^. teSpar) . path arPath . expect2xx)
+
+requestAccessVerdict :: HasCallStack
+                     => Bool                                   -- ^ is the verdict granted?
+                     -> (SAML.IdPId -> TestSpar ResponseLBS)   -- ^ raw authnreq
+                     -> TestSpar ( UserId
+                                 , SAML.ResponseVerdict
+                                 , URI                         -- ^ location header
+                                 , [(SBS, SBS)]                -- ^ query params
+                                 )
+requestAccessVerdict isGranted mkAuthnReq = do
+  env <- ask
+  let uid     = env ^. teUserId
+      idp     = env ^. teIdP
+      idpid   = idp ^. SAML.idpId
+      tenant  = idp ^. SAML.idpMetadata . SAML.edIssuer
+      subject = SAML.opaqueNameID "blee"
+      uref    = SAML.UserRef tenant subject
+  runSpar $ Spar.insertUser uref uid
+  authnreq :: SAML.FormRedirect SAML.AuthnRequest <- do
+    raw <- mkAuthnReq idpid
+    bdy <- maybe (error "authreq") pure $ responseBody raw
+    either (error . show) pure $ Servant.mimeUnrender (Servant.Proxy @SAML.HTML) bdy
+  spmeta <- getTestSPMetadata
+  authnresp <- do
+    let mk :: SAML.FormRedirect SAML.AuthnRequest -> TestSpar SAML.AuthnResponse
+        mk (SAML.FormRedirect _ req) = do
+          SAML.SignedAuthnResponse (XML.Document _ el _) <- runSimpleSP $ SAML.mkAuthnResponse SAML.sampleIdPPrivkey idp spmeta req True
```

Comment:
Lines longer than 78 chars break my screen :(

Reply:
```suggestion
          SAML.SignedAuthnResponse (XML.Document _ el _) <- runSimpleSP $
            SAML.mkAuthnResponse SAML.sampleIdPPrivkey idp spmeta req True
```

Reply:
I just did this to try out the suggest-code-changes github feature.  Is it ok if I am going to ignore this in the future?  78 columns is not enough!  At all!